[Bug]: Enabling bootimg causes a compilation error

### CheckBox

- [x] I have searched the issue list and found no similar issues.
- [x] I use the official action instead of someone else's
- [x] I have read the FAQ section in README.md

### Configuration

      with:
          kernel-url: https://github.com/Fyj69/kernel_oneplus_sdm845
          kernel-dir: mmp
          kernel-branch: oos/11.0
          config: sdm845-perf_defconfig
          arch: arm64
          aosp-gcc: true
          aosp-clang: true
          android-version: 12
          aosp-clang-version: r416183b
          ksu: true
          ksu-version: next
          ksu-other: true
          ksu-url: https://github.com/rifsxd/KernelSU-Next
          nethunter: true
          nethunter-patch: true
          bbrplus: true
          bootimg-url: https://github.com/Fyj69/patch/raw/main/boot/boot.img

### Describe the bug

Packaging boot.img
  
  10/24 22:38:22 [NOTICE] Downloading 1 item(s)
  
  10/24 22:38:22 [NOTICE] CUID#7 - Redirecting to https://raw.githubusercontent.com/Shubhamvis98/AIK/4ac321dfd48e16344e6146c505708aa720ff0bb3/bin/magiskboot_x86
  
  10/24 22:38:22 [NOTICE] Download complete: /home/runner/work/Zy-Action/Zy-Action/kernel/mmp/split/magiskboot
  
  Download Results:
  gid   |stat|avg speed  |path/URI
  ======+====+===========+=======================================================
  e1c214|OK  |   6.4MiB/s|/home/runner/work/Zy-Action/Zy-Action/kernel/mmp/split/magiskboot
  
  Status Legend:
  (OK):download completed.
  
  10/24 22:38:22 [NOTICE] Downloading 1 item(s)
  
  10/24 22:38:22 [NOTICE] CUID#7 - Redirecting to https://raw.githubusercontent.com/Fyj69/patch/main/boot/boot.img
  [#a87567 160KiB/64MiB(0%) CN:1 DL:4.6MiB ETA:13s]
  
  10/24 22:38:23 [NOTICE] Download complete: /home/runner/work/Zy-Action/Zy-Action/kernel/mmp/split/boot.img
  
  Download Results:
  gid   |stat|avg speed  |path/URI
  ======+====+===========+=======================================================
  a87567|OK  |   432MiB/s|/home/runner/work/Zy-Action/Zy-Action/kernel/mmp/split/boot.img
  
  Status Legend:
  (OK):download completed.
  Error: Process completed with exit code 1.

### Screenshots

_No response_

### How to reproduce

_No response_

### Expected behavior

_No response_

### Error Logs

_No response_

### Additional context

_No response_

--- a/action.yml
+++ b/action.yml
@@ -560,9 +560,9 @@ runs:
                 *) error "Unknown CPU architecture for this device!" ;;
             esac
             aria2c "${{ inputs.bootimg-url }}" -o boot.img || error "Failed to download boot.img"
-            ./magiskboot unpack boot.img >/dev/null 2>&1 || error "Failed to unpack boot.img"
-            rm -f kernel
-            FMT=$(./magiskboot decompress < ../out/arch/${{ inputs.arch }}/boot/Image 2>/dev/null | head -n 1 | cut -d ' ' -f 2)
+            ./magiskboot unpack boot.img > nohup.out 2>&1 # nohup is not working in github ci!
+            export FMT=$(cat nohup.out | grep "KERNEL_FMT" | awk '{gsub("\\[", "", $2); gsub("\\]", "", $2); print $2}')
+            rm -rf kernel
             if [ "$FMT" == "raw" ]; then
                 if ! find ../out/arch/${{ inputs.arch }}/boot -name "Image" ! -name "*-*" -exec cp -v {} kernel \; 2>/dev/null; then
                     error "Failed to find kernel Image"
@@ -578,7 +578,7 @@ runs:
                     fi
                 fi
             fi
-            ./magiskboot repack boot.img || error "Failed to repack boot.img"
+            ./magiskboot repack boot.img
             rm -f boot.img
             mkdir -p ../../../build
             find . -name "*.img" -exec mv -v {} ../../../build/boot.img \;

